File: src/cell.js
exports.Cell = function(x, y){
	this.x = x;
	this.y = y;	
	//this.midinote;
};

exports.Cell.prototype.fromMIDINote = function(midinote){
	this.y = Math.floor( (midinote[1] - 36 ) / 8);
	this.x = (midinote[1] - 36 )% 8;
};

exports.Cell.prototype.toMIDINote = function(){
	return (Math.floor(this.x + 36) / 8) + ((this.y + 36) % 8);
};

exports.Cell.prototype.equals = function(other){
	if (this.y === other.y && this.x === other.x) return true;

	return false;
};

File: teensy/gridquencer/Cell.cpp
#include "Cell.h"

Cell::Cell(char c_x, char c_y){
  _x = c_x;
  _y = c_y;
}

Cell::Cell(){
  _x = 0;
  _y = 0;
}

bool operator== (const Cell& c1, const Cell& c2){
  return (c1._x == c2._x && c1._y == c2._y);
}

bool operator!= (const Cell& c1, const Cell& c2){
  return (c1._x != c2._x || c1._y != c2._y);
}
